vllm/attention/backends/flashinfer.py

```diff
@@ -119,6 +125,35 @@ def get_fp8_dtype_for_flashinfer(kv_cache_dtype: str) -> torch.dtype:
         else:
             raise ValueError(f"Unrecognized FP8 dtype: {kv_cache_dtype}")
 
+    @staticmethod
+    def use_trtllm_decode_attention(batch_size: int,
+                                    max_seq_len: int,
+                                    kv_cache_dtype: str = "auto") -> bool:
+        # Check if environment variable is explicitly set
+        if FlashInferBackend.cached_use_trtllm is not None:
+            return FlashInferBackend.cached_use_trtllm
+        env_value = envs.VLLM_USE_TRTLLM_DECODE_ATTENTION
+        if env_value is not None:
+            logger.info_once("VLLM_USE_TRTLLM_DECODE_ATTENTION is set to %s",
+                             env_value)
+            # Environment variable is set - respect it
+            no_use_trtllm = env_value == "0"
+            if not no_use_trtllm:
+                logger.info_once(
+                    "VLLM_USE_TRTLLM_DECODE_ATTENTION is set to 1, "
+                    "using TRTLLM decode attention.")
+            FlashInferBackend.cached_use_trtllm = not no_use_trtllm
+        else:
+            # Environment variable not set - use auto-detection
+            use_trtllm = (is_sm100a_supported(torch.device("cuda"))
+                          and batch_size <= 256 and max_seq_len < 131072
+                          and kv_cache_dtype == "auto")
```

Comment:
Shouldn't we still update this value based on batch_size and max_seq_len for each forward pass? It seems like once we set `FlashInferBackend.cached_use_trtllm` once, the early exit will always trigger